### PR TITLE
xorg.*: update to latest version

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 241220` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>xorg.libAppleWM</li>
    <li>xorg.xf86videoark</li>
    <li>xorg.xf86videogeode</li>
    <li>xorg.xf86videoglide</li>
    <li>xorg.xf86videoglint</li>
    <li>xorg.xf86videoi128</li>
    <li>xorg.xf86videonewport</li>
    <li>xorg.xf86videos3virge</li>
    <li>xorg.xf86videosavage</li>
    <li>xorg.xf86videotga</li>
    <li>xorg.xf86videov4l</li>
    <li>xorg.xf86videovoodoo</li>
    <li>xorg.xf86videowsfb</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>xorg.xf86inputkeyboard</li>
  </ul>
</details>
<details>
  <summary>258 packages built:</summary>
  <ul>
    <li>xorg.appres</li>
    <li>xorg.bdftopcf</li>
    <li>xorg.bitmap</li>
    <li>xorg.editres</li>
    <li>xorg.encodings</li>
    <li>xorg.fontadobe100dpi</li>
    <li>xorg.fontadobe75dpi</li>
    <li>xorg.fontadobeutopia100dpi</li>
    <li>xorg.fontadobeutopia75dpi</li>
    <li>xorg.fontadobeutopiatype1</li>
    <li>xorg.fontarabicmisc</li>
    <li>xorg.fontbh100dpi</li>
    <li>xorg.fontbh75dpi</li>
    <li>xorg.fontbhlucidatypewriter100dpi</li>
    <li>xorg.fontbhlucidatypewriter75dpi</li>
    <li>xorg.fontbhttf</li>
    <li>xorg.fontbhtype1</li>
    <li>xorg.fontbitstream100dpi</li>
    <li>xorg.fontbitstream75dpi</li>
    <li>xorg.fontbitstreamspeedo</li>
    <li>xorg.fontbitstreamtype1</li>
    <li>xorg.fontcronyxcyrillic</li>
    <li>xorg.fontcursormisc</li>
    <li>xorg.fontdaewoomisc</li>
    <li>xorg.fontdecmisc</li>
    <li>xorg.fontibmtype1</li>
    <li>xorg.fontisasmisc</li>
    <li>xorg.fontjismisc</li>
    <li>xorg.fontmicromisc</li>
    <li>xorg.fontmisccyrillic</li>
    <li>xorg.fontmiscethiopic</li>
    <li>xorg.fontmiscmeltho</li>
    <li>xorg.fontmiscmisc</li>
    <li>xorg.fontmuttmisc</li>
    <li>xorg.fontschumachermisc</li>
    <li>xorg.fontscreencyrillic</li>
    <li>xorg.fontsonymisc</li>
    <li>xorg.fontsunmisc</li>
    <li>xorg.fonttosfnt</li>
    <li>xorg.fontwinitzkicyrillic</li>
    <li>xorg.fontxfree86type1</li>
    <li>xorg.iceauth</li>
    <li>xorg.ico</li>
    <li>xorg.imake</li>
    <li>xorg.libFS</li>
    <li>xorg.libICE</li>
    <li>xorg.libICE.dev</li>
    <li>xorg.libICE.doc</li>
    <li>xorg.libSM</li>
    <li>xorg.libSM.dev</li>
    <li>xorg.libSM.doc</li>
    <li>xorg.libWindowsWM</li>
    <li>xorg.libX11</li>
    <li>xorg.libX11.dev</li>
    <li>xorg.libX11.man</li>
    <li>xorg.libXScrnSaver</li>
    <li>xorg.libXTrap</li>
    <li>xorg.libXau</li>
    <li>xorg.libXau.dev</li>
    <li>xorg.libXaw</li>
    <li>xorg.libXaw.dev</li>
    <li>xorg.libXaw.devdoc</li>
    <li>xorg.libXcomposite</li>
    <li>xorg.libXcomposite.dev</li>
    <li>xorg.libXcursor</li>
    <li>xorg.libXcursor.dev</li>
    <li>xorg.libXdamage</li>
    <li>xorg.libXdamage.dev</li>
    <li>xorg.libXdmcp</li>
    <li>xorg.libXdmcp.dev</li>
    <li>xorg.libXdmcp.doc</li>
    <li>xorg.libXext</li>
    <li>xorg.libXext.dev</li>
    <li>xorg.libXext.doc</li>
    <li>xorg.libXext.man</li>
    <li>xorg.libXfixes</li>
    <li>xorg.libXfixes.dev</li>
    <li>xorg.libXfont</li>
    <li>xorg.libXfont.dev</li>
    <li>xorg.libXfont2</li>
    <li>xorg.libXft</li>
    <li>xorg.libXft.dev</li>
    <li>xorg.libXi</li>
    <li>xorg.libXi.dev</li>
    <li>xorg.libXi.doc</li>
    <li>xorg.libXi.man</li>
    <li>xorg.libXinerama</li>
    <li>xorg.libXinerama.dev</li>
    <li>xorg.libXmu</li>
    <li>xorg.libXmu.dev</li>
    <li>xorg.libXmu.doc</li>
    <li>xorg.libXp</li>
    <li>xorg.libXp.dev</li>
    <li>xorg.libXpm</li>
    <li>xorg.libXpm.bin</li>
    <li>xorg.libXpm.dev</li>
    <li>xorg.libXpresent</li>
    <li>xorg.libXrandr</li>
    <li>xorg.libXrandr.dev</li>
    <li>xorg.libXrender</li>
    <li>xorg.libXrender.dev</li>
    <li>xorg.libXrender.doc</li>
    <li>xorg.libXres</li>
    <li>xorg.libXres.dev</li>
    <li>xorg.libXres.devdoc</li>
    <li>xorg.libXt</li>
    <li>xorg.libXt.dev</li>
    <li>xorg.libXt.devdoc</li>
    <li>xorg.libXtst</li>
    <li>xorg.libXv</li>
    <li>xorg.libXv.dev</li>
    <li>xorg.libXv.devdoc</li>
    <li>xorg.libXvMC</li>
    <li>xorg.libXvMC.dev</li>
    <li>xorg.libXvMC.doc</li>
    <li>xorg.libXxf86dga</li>
    <li>xorg.libXxf86misc</li>
    <li>xorg.libXxf86vm</li>
    <li>xorg.libXxf86vm.dev</li>
    <li>xorg.libdmx</li>
    <li>xorg.libfontenc</li>
    <li>xorg.libpciaccess</li>
    <li>xorg.libxcb</li>
    <li>xorg.libxcb.dev</li>
    <li>xorg.libxcb.doc</li>
    <li>xorg.libxcb.man</li>
    <li>xorg.libxkbfile</li>
    <li>xorg.libxkbfile.dev</li>
    <li>xorg.libxshmfence</li>
    <li>xorg.libxshmfence.dev</li>
    <li>xorg.listres</li>
    <li>xorg.makedepend</li>
    <li>xorg.mkfontdir (xorg.mkfontscale)</li>
    <li>xorg.oclock</li>
    <li>xorg.sessreg</li>
    <li>xorg.setxkbmap</li>
    <li>xorg.smproxy</li>
    <li>xorg.transset</li>
    <li>xorg.twm</li>
    <li>xorg.utilmacros</li>
    <li>xorg.viewres</li>
    <li>xorg.wrapWithXFileSearchPathHook</li>
    <li>xorg.x11perf</li>
    <li>xorg.xauth</li>
    <li>xorg.xbacklight</li>
    <li>xorg.xbitmaps</li>
    <li>xorg.xcalc</li>
    <li>xorg.xcbproto</li>
    <li>xorg.xcbutil</li>
    <li>xorg.xcbutil.dev</li>
    <li>xorg.xcbutilcursor</li>
    <li>xorg.xcbutilcursor.dev</li>
    <li>xorg.xcbutilerrors</li>
    <li>xorg.xcbutilerrors.dev</li>
    <li>xorg.xcbutilimage</li>
    <li>xorg.xcbutilimage.dev</li>
    <li>xorg.xcbutilkeysyms</li>
    <li>xorg.xcbutilkeysyms.dev</li>
    <li>xorg.xcbutilrenderutil</li>
    <li>xorg.xcbutilrenderutil.dev</li>
    <li>xorg.xcbutilwm</li>
    <li>xorg.xcbutilwm.dev</li>
    <li>xorg.xclock</li>
    <li>xorg.xcmsdb</li>
    <li>xorg.xcompmgr</li>
    <li>xorg.xconsole</li>
    <li>xorg.xcursorgen</li>
    <li>xorg.xcursorthemes</li>
    <li>xorg.xdm</li>
    <li>xorg.xdpyinfo</li>
    <li>xorg.xdriinfo</li>
    <li>xorg.xev</li>
    <li>xorg.xeyes</li>
    <li>xorg.xf86inputevdev</li>
    <li>xorg.xf86inputevdev.dev</li>
    <li>xorg.xf86inputjoystick</li>
    <li>xorg.xf86inputlibinput</li>
    <li>xorg.xf86inputlibinput.dev</li>
    <li>xorg.xf86inputmouse</li>
    <li>xorg.xf86inputsynaptics</li>
    <li>xorg.xf86inputsynaptics.dev</li>
    <li>xorg.xf86inputvmmouse</li>
    <li>xorg.xf86inputvoid</li>
    <li>xorg.xf86videoamdgpu</li>
    <li>xorg.xf86videoapm</li>
    <li>xorg.xf86videoast</li>
    <li>xorg.xf86videoati</li>
    <li>xorg.xf86videochips</li>
    <li>xorg.xf86videocirrus</li>
    <li>xorg.xf86videodummy</li>
    <li>xorg.xf86videofbdev</li>
    <li>xorg.xf86videoi740</li>
    <li>xorg.xf86videointel</li>
    <li>xorg.xf86videomga</li>
    <li>xorg.xf86videoneomagic</li>
    <li>xorg.xf86videonouveau</li>
    <li>xorg.xf86videonv</li>
    <li>xorg.xf86videoomap</li>
    <li>xorg.xf86videoopenchrome</li>
    <li>xorg.xf86videoqxl</li>
    <li>xorg.xf86videor128</li>
    <li>xorg.xf86videorendition</li>
    <li>xorg.xf86videosiliconmotion</li>
    <li>xorg.xf86videosis</li>
    <li>xorg.xf86videosisusb</li>
    <li>xorg.xf86videosuncg6</li>
    <li>xorg.xf86videosunffb</li>
    <li>xorg.xf86videosunleo</li>
    <li>xorg.xf86videotdfx</li>
    <li>xorg.xf86videotrident</li>
    <li>xorg.xf86videovboxvideo</li>
    <li>xorg.xf86videovesa</li>
    <li>xorg.xf86videovmware</li>
    <li>xorg.xf86videoxgi</li>
    <li>xorg.xfd</li>
    <li>xorg.xfontsel</li>
    <li>xorg.xfs</li>
    <li>xorg.xfsinfo</li>
    <li>xorg.xgamma</li>
    <li>xorg.xgc</li>
    <li>xorg.xhost</li>
    <li>xorg.xinit</li>
    <li>xorg.xinput</li>
    <li>xorg.xkbcomp</li>
    <li>xorg.xkbevd</li>
    <li>xorg.xkbprint</li>
    <li>xorg.xkbutils</li>
    <li>xorg.xkeyboardconfig</li>
    <li>xorg.xkill</li>
    <li>xorg.xload</li>
    <li>xorg.xlsatoms</li>
    <li>xorg.xlsclients</li>
    <li>xorg.xlsfonts</li>
    <li>xorg.xmag</li>
    <li>xorg.xmessage</li>
    <li>xorg.xmodmap</li>
    <li>xorg.xmore</li>
    <li>xorg.xorgcffiles</li>
    <li>xorg.xorgdocs</li>
    <li>xorg.xorgproto</li>
    <li>xorg.xorgserver</li>
    <li>xorg.xorgserver.dev</li>
    <li>xorg.xorgsgmldoctools</li>
    <li>xorg.xpr</li>
    <li>xorg.xprop</li>
    <li>xorg.xrandr</li>
    <li>xorg.xrdb</li>
    <li>xorg.xrefresh</li>
    <li>xorg.xset</li>
    <li>xorg.xsetroot</li>
    <li>xorg.xsm</li>
    <li>xorg.xstdcmap</li>
    <li>xorg.xtrans</li>
    <li>xorg.xtrap</li>
    <li>xorg.xvinfo</li>
    <li>xorg.xwd</li>
    <li>xorg.xwininfo</li>
    <li>xorg.xwud</li>
  </ul>
</details>

File: pkgs/servers/x11/xorg/overrides.nix [hunk 1]
```diff
@@ -419,7 +419,11 @@ self: super:
     };
   });
 
-  xf86inputkeyboard = brokenOnDarwin super.xf86inputkeyboard; # never worked: https://hydra.nixos.org/job/nixpkgs/trunk/xorg.xf86inputkeyboard.x86_64-darwin
+  xf86inputkeyboard = super.xf86inputkeyboard.overrideAttrs (attrs: {
+    meta = attrs.meta // {
+      platforms = lib.platforms.freebsd ++ lib.platforms.netbsd ++ lib.platforms.openbsd;
+    };
+  });
```
Comment (vcunat) [inline pkgs/servers/x11/xorg/overrides.nix:426]: Why did you drop it from `*-linux`?

Comment (NickCao) [reply pkgs/servers/x11/xorg/overrides.nix:426]: Because it explicitly says that linux is not a supported platform in the configure script.

Comment (NickCao) [reply pkgs/servers/x11/xorg/overrides.nix:426]: Ref: https://gitlab.freedesktop.org/xorg/driver/xf86-input-keyboard/-/blob/master/configure.ac#L61

**Pull request merged**